clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -634,6 +639,10 @@ protected void checkLongRunningTaskState() throws InterruptedException {
         case PAUSE_TRANSITION_FROM_STANDBY_TO_LEADER:
           Store store = storeRepository.getStoreOrThrow(storeName);
           if (!store.isMigrationDuplicateStore()) {
+            heartbeatMonitoringService.updateLagMonitor(
```

Comment:
Is this right?  We only want to set up this monitor when the state transition to leader is done right?

....or.... does the isReadyToServe filter prevent this from being an issue?

Reply:
It is not what we are doing right now though. What we are doing today is when it is appointed to be LEADER from Helix, we start tracking it as leader lag. 
With the offline chat, I am a bit hesitated - maybe it is simpler if we do as what you said? The lag is there anyway, but we have fewer lag action step in the transition